geo/src/algorithm/relate/geomgraph/geometry_graph.rs

```diff
@@ -35,7 +34,7 @@ where
 {
     arg_index: usize,
     parent_geometry: GeometryCow<'a, F>,
-    tree: Option<Rc<RTree<Segment<F>>>>,
+    tree: Option<RTree<Segment<F>>>,
```

Comment:
There are some positive changes in this PR, but there is a serious perf regression with:
```
 relate prepared polygons
                        time:   [142.78 ms 143.14 ms 143.63 ms]
                        change: [+192.56% +193.59% +194.76%] (p = 0.00 < 0.05)
                        Performance has regressed.
```

I haven't zeroed in on the cause yet - but this line is suspect. The point of the `Rc<Tree>` is that it was cheap to clone. It seems like we've removed that and now have a "cache the tree and invalidate it as necessary" approach.

Note there are also some nice performance wins to the *non*-prepared geometry operations in this PR, but I personally don't think it's worth it as phrased.

Reply:
Ooof yeah that regression is a show-stopper at the moment.